regression/scripts/battery/battery_cell_discharge_tests.py

```diff
@@ -0,0 +1,491 @@
+
+    # base segment
+    base_segment                                             = Segments.Segment()
+    ones_row                                                 = base_segment.state.ones_row
+    base_segment.state.numerics.number_control_points        = 20
+    base_segment.process.iterate.initials.initialize_battery = SUAVE.Methods.Missions.Segments.Common.Energy.initialize_battery  
+    base_segment.process.iterate.conditions.stability        = SUAVE.Methods.skip
+    base_segment.process.finalize.post_process.stability     = SUAVE.Methods.skip 
+    base_segment.process.iterate.conditions.aerodynamics     = SUAVE.Methods.skip
+    base_segment.process.finalize.post_process.aerodynamics  = SUAVE.Methods.skip     
+    base_segment.process.iterate.conditions.planet_position  = SUAVE.Methods.skip
+    
+   
+    bat                                                      = vehicle.networks.battery_cell.battery    
+    base_segment.max_energy                                  = bat.max_energy
+    base_segment.charging_SOC_cutoff                         = bat.cell.charging_SOC_cutoff 
+    base_segment.charging_current                            = bat.charging_current
+    base_segment.charging_voltage                            = bat.charging_voltage 
+    base_segment.initial_battery_resistance_growth_factor    = 1
+    base_segment.initial_battery_capacity_fade_factor        = 1            
+    base_segment.temperature_deviation                       = 15
+    discharge_time                                           = 0.9 * (mAh/1000)/current * Units.hrs
+    
+    if battery_chemistry == 'LFP':
+        segment                                             = Segments.Ground.Battery_Charge_Discharge(base_segment)
+        segment.tag                                         = "LFP_Discharge" 
+        segment.analyses.extend(analyses.base)       
+        segment.time                                        = discharge_time 
+        segment.battery_pack_temperature                    = 300  
+        segment.ambient_temperature                         = 300   
+        segment.initial_battery_charge_throughput           = 0
+        segment.battery_energy                              = bat.max_energy * 1.
+        segment = vehicle.networks.battery_cell.add_unknowns_and_residuals_to_segment(segment)    
+        mission.append_segment(segment)         
+        
+        # Charge Model 
+        segment                                             = Segments.Ground.Battery_Charge_Discharge(base_segment)     
+        segment.tag                                         = 'LFP_Charge'  
+        segment.battery_discharge                           = False 
+        segment.analyses.extend(analyses.base)        
+        segment = vehicle.networks.battery_cell.add_unknowns_and_residuals_to_segment(segment)      
+        mission.append_segment(segment) 
+        
+        
+    if battery_chemistry == 'NCA':
```

Comment:
looks like you only need the if-else statement for the segment tags; pull out all the common code to outside the if-else statement

Reply:
corrected